contracts/contracts/cairo/validator/Price_Consumer.cairo

```diff
@@ -0,0 +1,71 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from cairo.ocr2.interfaces.IAggregator import IAggregator, Round
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.math_cmp import is_nn
+from starkware.cairo.common.bool import TRUE, FALSE
+from starkware.starknet.common.syscalls import (
+    get_block_timestamp,
+)
+
+@storage_var
+func uptime_feed_address_() -> (address : felt):
+end
+
+@storage_var
+func aggregator_address_() -> (address : felt):
+end
+
+@constructor
+func constructor{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr,
+}(uptime_feed_address : felt, aggregator_address : felt):
+    
+    assert_not_zero(uptime_feed_address)
+    assert_not_zero(aggregator_address)
+
+    uptime_feed_address_.write(uptime_feed_address)
+    aggregator_address_.write(aggregator_address)
+    return ()
+end
+
+@view
+func get_latest_price{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr,
+}() -> (round : felt):
+    let (sequencer_is_up) = check_sequencer_state()
+    with_attr error_message("L2 sequencer down: Chainlink feeds are not being updated"):
+        assert sequencer_is_up = TRUE
+    end
+    let (aggregator_address) = aggregator_address_.read()
+    let (round : Round) = IAggregator.latest_round_data(contract_address=aggregator_address)
+    return (round=round.answer)
+end
+
+@external
+func check_sequencer_state{
```

Comment:
Please add some comments/docs to the function explaining what are you exactly checking here.

Reply:
I'd suggest you restructure this function as `assert_sequencer_healthy` and then inside error in three cases with different error messages:

- L2 Sequencer is down, report ok
- L2 Sequencer is down, report stale
- L2 Sequencer is up, report stale

Your current logic assumes 1h heartbeat, after which the report is considered stale. Let's default that to 60 sec for these example/demo purposes.